src/csm_testing/lib/iscsi_common.py

```diff
@@ -0,0 +1,289 @@
+# MIT License
+#
+# (C) Copyright 2025 Hewlett Packard Enterprise Development LP
+#
+# Permission is hereby granted, free of charge, to any person obtaining a
+# copy of this software and associated documentation files (the "Software"),
+# to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense,
+# and/or sell copies of the Software, and to permit persons to whom the
+# Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included
+# in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+"""
+The script has common functions being used by iscsi tests scripts
+"""
+import sys
+import subprocess
+import re
+from packaging import version
+
+def run_command(command):
+    """
+    Function to run a given command
+    Args:
+        command(str): Command to be executed
+    Returns:
+        stdout/stderr(str),returncode(int)
+    """
+    try:
+        result = subprocess.run(
+            command,
+            shell=True,
+            check=True,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+            universal_newlines=True,
+        )
+        return result.stdout.strip(), result.returncode
+    except subprocess.CalledProcessError as err:
+        if err.stderr.strip():
+            return err.stderr.strip(), err.returncode
+        return err.stdout.strip(), err.returncode
+
+
+def compare_versions(version1, version2):
+    """
+    Function to compare given two versions and print the result info.
+    Args:
+        version1(int/float), version2(int/float) : versions to compare
+    Returns:
+        Bool
+    """
+    ver_1 = version.parse(version.Version(version1).base_version)
+    ver_2 = version.parse(version2)
+
+    if ver_1 < ver_2:
+        return False
+    return True
+
+
+def iscsi_prechecks_failed():
```

Comment:
Calling `sys.exit` technically raises an exception, so this function actually never returns -- hence `NoReturn` is the proper type annotation.
```suggestion
def iscsi_prechecks_failed() -> NoReturn:
```